### PR TITLE
feat(pytest): Migrate legacy `groups` unit test to Pytest

### LINKED ISSUE #806: Pytest: Migrate feature test `groups.chroot`
**What would you like to be added**:
The following test should be migrated to a pytest unit:
* [features/base/test/groups.chroot](https://github.com/gardenlinux/gardenlinux/blob/main/features/base/test/groups.chroot)

Thereby, there are already some pytest units ([here](https://github.com/gardenlinux/gardenlinux/tree/main/tests/helper/tests)), which can be used as blueprints for the migrated test.

**Why is this needed**:
We want to migrate each feature test to a corresponding pytest. Once we have done this, we can make the `chroot` and `kvm` platform tests the new default tests for Garden Linux builds.

### PR DESCRIPTION
**Pull request by gyptazy**
feat(pytest): Migrate legacy `groups` unit test to Pytest

**How to categorize this PR?**

/kind test
/area os
/os garden-linux

**What this PR does / why we need it**:
Migrate legacy `groups` unit test to Pytest

**Which issue(s) this PR fixes**:
Fixes #806

**Release note**:

```feature user
NONE
```

### CODE REVIEW

File: tests/helper/tests/users.py [hunk 1]
```diff
@@ -51,3 +55,24 @@ def users(client):
...
+
+
+def _has_user_sudo_cmd(client, user):
+    """ Check if user has any sudo permissions """
+    # Execute command on remote platform
+    cmd = f"sudo -l -U {user}"
+    out = utils.execute_remote_command(client, cmd)
+
+    # Write each line as output in list
+    output_lines = []
```
Comment (nanory) [inline tests/helper/tests/users.py:67]: why not simply calling it like this?
```
output_lines = out.split("\n")
```

File: tests/helper/tests/users.py [hunk 2]
```diff
@@ -51,3 +55,24 @@ def users(client):
...
+    output_lines = [] 
+    for line in out.split("\n"):
+       output_lines.append(line)
+
+    # Check if there is enough content in our list
+    # and validate if there are related sudo commands
+    sudo_cmd = False
+    if len(output_lines) > 3:
+        if "may run the following commands on" in output_lines[-2]:
+           sudo_cmd = output_lines[-1]
```
Comment (nanory) [inline tests/helper/tests/users.py:76]: I don't know if this is robust enough. What if a user has the privileges to execute multiple commands via sudo. The rule set can be longer than just having one entry, which this implementation assumes. In this case `[-2]` and `[-1]` does not work properly.

File: tests/helper/tests/users.py (hunk 1 above)
Comment (gyptazy) [reply tests/helper/tests/users.py:67]: at this point, yes. But the loop will be needed later. tbh, we can reverse the order (as we expect the content within the last lines).

File: tests/helper/tests/users.py (hunk 2 above)
Comment (gyptazy) [reply tests/helper/tests/users.py:76]: true.

**Review by chrinorse — APPROVED** [review]
lgtm

**Pull request merged**